### ISSUE TITLE
Cisco ASA NF_F_INGRESS_ACL_ID and NF_F_EGRESS_ACL_ID field support

### ISSUE THREAD

**Issue by jstopinsek** [logstash-plugins/logstash-codec-netflow#20]
Please consider to support two Cisco ASA NetFlow fields NF_F_INGRESS_ACL_ID (33000) and NF_F_EGRESS_ACL_ID (33001) which are described in this document:

http://www.cisco.com/c/en/us/td/docs/security/asa/asa93/netflow/netflow.html

This two fields are a sequence of bytes.
The 12-byte raw ACL ID must be divided into its three constituent parts, as follows:

    The first four bytes are the ACL Name ID
    The next four bytes are the ACL Entry ID (ACE)/Object-Group ID
    The final four bytes are the Extended ACL Entry ID

And at the end field values should look like:
0xa231c4d3 - 0xb34531ad - 0x00000000

![cisco-asa-show-access-list](https://cloud.githubusercontent.com/assets/14274675/9849959/b5bf653a-5aef-11e5-8463-99fc22d189bd.png)

------------------------------------------------------------------
I'm also posting an example with my configuration:

Parameters in netflow.yaml:

33000:
- 12
- :ingress_acl_id

Elasticsearch template parameters:

"ingress_acl_id":{
              "index":"not_analyzed","type":"string","norms":{"enabled":false},"doc_values":true,
              "fields":{"raw":{"index":"not_analyzed","type":"string","norms":{"enabled":false},"doc_values":true,"ignore_above":256}}

Elasticsearch field content:
netflow.ingress_acl_id:47141090178447234867545505792

tshark field content:
Ingress ACL ID: 985232233d89704900000000

Tshark field value is correct and should be in elasticsearch also.
Maybe it would be nicer if it was devided by dash: 98523223-3d897049-00000000 or even written in separate fields.

**Comment by jorritfolmer** [comment]
Could you try to test PR #24 to see if it fixes this?

**Comment by broknbottle** [comment]
I don't believe it's working as I am receiving an error that the numeric value is too large to be parsed,

"status"=>400, "error"=>{"type"=>"mapper_parsing_exception", "reason"=>"failed to parse [netflow.ingress_acl_id]", "caused_by"=>{"type"=>"json_parse_exception", "reason"=>"Numeric value (40304685089359864798539415552) out of range of long (-9223372036854775808 - 9223372036854775807)\n at [Source: org.elasticsearch.common.io.stream.InputStreamStreamInput@5830c090; line: 1, column: 565]"}}}}, :level=>:warn}

**Comment by jorritfolmer** [comment]
It looks like Elasticsearch is unable to parse the rather large numeric value that Logstash provides.
However, I think having Logstash output it as a string field would be more appropriate since `the 12-byte raw ACL ID must be divided into its three constituent parts`

**Comment by jorritfolmer** [comment]
I've updated PR #24 to decode the ingress and egress acl fields like this:

````
[snip]
             "ingress_acl_id" => "0f8e7ff3-fc1a030f-00000000",
              "egress_acl_id" => "00000000-00000000-00000000",

[snip]
````

**Comment by broknbottle** [comment]
@jorritfolmer thanks! I will test this out.

**Comment by ssi0202** [comment]
any word on how this worked out?

**Comment by jorritfolmer** [comment]
Closed with merge of #24
